docs/DeveloperGuide.md

```diff
@@ -285,134 +285,95 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
+* 4a. The given index is invalid.
+    * a1. MassLinkers prompts for another index.
+    * Step a1 is repeated until a valid index is given. \
+  Use case resumes from step 3.
 
-* 3a. The given index is invalid.
-    * 3a1. MassLinkers shows an error message.
-      \
-      Use case resumes at step 2.
-
-
-* a. At any time, Student chooses to close MassLinkers.
-    * a1. MassLinkers updates the written storage file.
-    * a2. MassLinkers closes the program.
+* *a. At any time, Student chooses to close MassLinkers.
+    * a1. MassLinkers updates the storage file.
       \
       Use case ends.
 
-**Use case 3: Search classmate via a tag**
+**Use case 3: Search classmates via a tag**
 
 **MSS**
 
-1. Student searches classmate with a particular tag.
-2. MassLinkers verifies syntax of the search command is valid.
-3. MassLinkers searches through the list of added tags.
-4. MassLinkers returns the list of classmates associated with the particular tag and display it through the GUI.
+1. Student requests the list of batchmates with a tag.
+2. MassLinkers searches through the list of added tags.
+3. MassLinkers returns the list of batchmates associated with the particular tag.
    \
    Use case ends.
 
 **Extensions**
-* 3a. MassLinkers detects an error in the sequence or syntax of entered command.
-    * 3a1. MassLinkers prompts error message.
-    * 3a2. MassLinkers requests for the correct command.
-    * 3a3. Student enters new command.
-      Steps 3a1-3a3 are repeated until the command entered are correct.
-      \
-      Use case resumes from step 4.
-
-
-* 4a. MassLinkers detects there is no such tag.
-    * 4a1. MassLinkers prompts error message.
-    * 4a2. MassLinkers requests for the correct command.
-    * 4a3. Student enters new command.
-      \
-      Steps 4a1-4a3 are repeated until the command entered are correct.
-      \
-      Use case resumes from step 4.
-
-
-* a. At any time, Student chooses to close MassLinkers.
-    * a1. MassLinkers updates the written storage file.
-    * a2. MassLinkers closes the program.
+* 2a. The list of tags is empty.
+  * a1. MassLinkers returns an empty list. 
+    \
+  Use case ends.
+* 2b. MassLinkers detects there is no such tag.
+  * b1. MassLinkers requests for another tag.
+  * Step b1 is repeated until a valid tag is found. \
+  Use case resumes from step 2.
+* *a. At any time, Student chooses to close MassLinkers.
+    * a1. MassLinkers updates the storage file.
       \
       Use case ends.
 
 
-**Use case 4: Add a tag to a classmate**
+**Use case 4: Add a tag to a batchmate**
 
 **MSS**
 
-1. Student chooses to add a tag on a classmate, could be himself or others.
-2. Student enters the add tag command with the associated classmate.
-3. MassLinkers verifies the syntax of the command.
-4. MassLinkers searches through the list of people for the selected classmate.
-5. MassLinkers adds tag to the classmate.
+1. Student chooses to add a tag to a batchmate (could be himself or others).
```

Comment:
I think the things mentioned in brackets (could be himself or others) could be removed to avoid confusion, as a student may or may not have him or herself added in MassLinkers.